core/builtin.py

```diff
@@ -540,11 +540,17 @@ def Dirs(argv, dir_stack):
   if arg.c:
     del dir_stack[:]
   elif arg.v:
-    for i, entry in enumerate(dir_stack):
-      print('%2d %s' % (i, entry))
+    temp_stack = dir_stack[:]
+    temp_stack.insert(0, os.getcwd())
+    for i, entry in enumerate(temp_stack):
+      print('%2d  %s' % (i, entry))
+    sys.stdout.flush()
   elif arg.p:
-    for entry in dir_stack:
+    temp_stack = dir_stack[:]
+    temp_stack.insert(0, os.getcwd())
```

Comment:
This could be written as:

```
dirs = [os.getcwd()]
dirs.extend(dir_stack)
```

In general it's more idiomatic to append to lists rather than insert into the middle.  Because insertion is `O(n)` while appending is constant time.

Reply:
No problem, I can make that change.